src/sentry/integrations/issue_alert_image_builder.py

```diff
@@ -0,0 +1,121 @@
+import logging
+
+import sentry_sdk
+
+from sentry.api import client
+from sentry.charts import backend as charts
+from sentry.charts.types import ChartType
+from sentry.integrations.slack.message_builder.time_utils import (
+    get_approx_start_time,
+    get_relative_time,
+)
+from sentry.issues.grouptype import (
+    PerformanceP95EndpointRegressionGroupType,
+    ProfileFunctionRegressionType,
+)
+from sentry.models.apikey import ApiKey
+from sentry.models.group import Group
+from sentry.snuba.referrer import Referrer
+from sentry.types.integrations import ExternalProviderEnum
+from sentry.utils import metrics
+from sentry.utils.performance_issues.detectors.utils import escape_transaction
+
+
+class IssueAlertImageBuilder:
+    def __init__(
+        self, group: Group, provider: ExternalProviderEnum, logger: logging.Logger
+    ) -> None:
+        self.group = group
+        self.provider = provider
+        self.logger = logger
+        self.cache_key = f"chartcuterie-image:{self.group.id}"
+        self.tags = {
+            "provider": self.provider,
+            "issue_category": self.group.issue_category,
+        }
+
+    def get_image_url(self) -> str | None:
+        try:
+            metrics.incr("chartcuterie.issue_alert.attempt", tags=self.tags)
+            if self.group.issue_type == PerformanceP95EndpointRegressionGroupType:
+                image_url = self._get_endpoint_regression_image_url()
+            elif self.group.issue_type == ProfileFunctionRegressionType:
+                image_url = self._get_function_regression_image_url()
+
+            if image_url:
+                metrics.incr("chartcuterie.issue_alert.success", tags=self.tags)
+                return image_url
+        except Exception as e:
+            self.logger.exception(
+                "issue_alert_chartcuterie_image.failed",
+                extra={"exception": e, "group_id": self.group.id},
+            )
+            sentry_sdk.capture_exception()
+        return None
+
+    def _get_endpoint_regression_image_url(self) -> str | None:
```

Comment:
Nothing is new below this line. All copy paste from image_block_builder

Reply:
Super helpful comment for reviewing, thank you for this haha

Reply:
Wish GH just had a "please mark this as moved" or something to help reviewers